pkg/jobrunaggregator/jobrunaggregatorlib/ci_data_client.go

```diff
@@ -112,11 +140,13 @@ LIMIT 1
 func (c *ciDataClient) GetLastJobRunWithDisruptionDataForJobName(ctx context.Context, jobName string) (*jobrunaggregatorapi.JobRunRow, error) {
 	// the JobRun.Name is always increasing, so we can sort by that name.  The starttime is based on the prowjob
 	// time and I don't think that is coordinated.
+	// the disruption jobrun table is now distinct, so we will use the disruption jobrun table as authoritative for what data should and should not
```

Comment:
This created a huge efficiency gain.  Apparently we were trying to gather backend disruption data for many job types that don't have it, so we effectively scanned the entire GCS bucket over and over.